src/components/popupWithForm.js

```diff
@@ -20,7 +21,9 @@ export default class PopupWithForm extends Popup {
   }
 
   setEventListeners() {
+    //modify to work with modals without inputs
     super.setEventListeners();
+    //
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`